test/h2_worker_test.exs

```diff
@@ -489,8 +491,9 @@ defmodule Sparrow.H2WorkerTest do
             context[:auth]
           )
 
-        {:ok, pid} = GenServer.start_link(Sparrow.H2Worker, config)
-        state = :sys.get_state(pid)
+        worker_pid = start_supervised!(Tools.h2_worker_spec(config))
+        :timer.sleep(100)
+        state = :sys.get_state(worker_pid)
```

Comment:
Do we use that state? Please make sure all compile warnings that are generated here are fixed.

Reply:
Yes, we do use it